asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -159,36 +252,65 @@ def to_sdf(self, filename: Union[str, Path], write_SD_attrs: bool = True) -> Non
         filename : Union[str, Path]
             Path to the SDF file
         write_SD_attrs : bool, optional
-            Whether to write out SD tags as attributes, by default True
+            Whether to write out attributes as SD tags, by default True
 
         """
         if write_SD_attrs:
-            self.flush_attrs_to_SD_data()
+            data_to_write = self.flush_attrs_to_SD_data()
+        else:
+            data_to_write = self.data
         # directly write out data
-        write_file_directly(filename, self.data)
+        write_file_directly(filename, data_to_write)
 
     def set_SD_data(self, data: dict[str, str]) -> None:
-        mol = sdf_string_to_oemol(self.data)
-        mol = set_SD_data(mol, data)
-        self.data = oemol_to_sdf_string(mol)
+        """
+        Set the SD data for the ligand, uses an update to overwrite existing data in line with
+        OpenEye behaviour
+        """
+        self.tags.update(data)
```

Comment:
what do we want to happen if you try to update the SD tags with a reserved field?

Reply:
Good spotting, added a check for that and a test.